src/jimgw/core/single_event/detector.py

```
@@ -586,7 +586,7 @@ def inject_signal(
         waveform_model,
         parameters: dict[str, float],
         is_zero_noise: bool = False,
-        rng_key: PRNGKeyArray = jax.random.PRNGKey(0),
+        rng_key: Key = jax.random.key(0),
```

Ruff (0.15.2)

[warning] Do not perform function call `jax.random.key` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)